src/r3d.imageio/r3dinput.cpp

```diff
@@ -0,0 +1,441 @@
+// Copyright Contributors to the OpenImageIO project.
+// SPDX-License-Identifier: Apache-2.0
+// https://github.com/AcademySoftwareFoundation/OpenImageIO
+
+// R3D SDK can be downloaded from the following site:
+// https://www.red.com/download/r3d-sdk
+//
+// The code has been tested with the version 8.5.1 installed in
+// /opt/R3DSDKv8_5_1 directory and setting up the variable
+// export R3DSDK_ROOT="/opt/R3DSDKv8_5_1"
+
+#include <algorithm>
+#include <cassert>
+#include <cstdio>
+
+#include <OpenImageIO/filesystem.h>
+#include <OpenImageIO/fmath.h>
+#include <OpenImageIO/imageio.h>
+#include <OpenImageIO/tiffutils.h>
+
+#include <R3DSDK.h>
+#include <R3DSDKCuda.h>
+#include <R3DSDKDefinitions.h>
+
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <cuda_runtime.h>
+#include <vector>
+
+#include <condition_variable>
+#include <functional>
+#include <iostream>
+#include <list>
+#include <map>
+#include <mutex>
+#include <thread>
+#include <time.h>
+#include <unistd.h>
+#include <vector>
+
+unsigned char*
+AlignedMalloc(size_t& sizeNeeded)
```

Comment:
I believe we already have an aligned_malloc and aligned_new in platform.h

Reply:
Good 2 know. Will modify the code accordingly.